src/main/java/app/view/SideMenu.java

```diff
@@ -21,26 +30,137 @@ public SideMenu(int width, int height)
+                {
+                    case "wall" ->
+                            {
+                                dr.setColour(Color.SANDYBROWN);
+                                dr.setFill(true);
+                            }
+                    case "shade" ->
+                            {
+                                dr.setColour(Color.LIGHTGRAY);
+                                dr.setFill(true);
+                            }
+                    case "glass" ->
+                            {
+                                dr.setColour(Color.LIGHTBLUE);
+                                dr.setFill(true);
+                            }
+                    case "tower" ->
+                            {
+                                dr.setColour(Color.OLIVE);
+                                dr.setFill(true);
+                            }
+                    case "teleport" ->
+                            {
+                                dr.setVector(new Vector(parseRect(x), parseRect(y)));
+                                dr.setColour(Color.OLIVEDRAB);
+                                dr.setFill(true);
+                            }
+                    case "spawnAreaGuards" ->
+                            {
+                                dr.setColour(Color.BLUE);
+                                dr.setFill(false);
+                            }
+                    case "spawnAreaIntruders" ->
+                            {
+                                dr.setColour(Color.RED);
+                                dr.setFill(false);
+                            }
+                    case "targetArea" ->
+                            {
+                                dr.setColour(Color.GOLD);
+                                dr.setFill(false);
+                            }
+                }
+                mb.run(dr);
+            });
             vbox.getChildren().add(furnType);
         }
 
         // Functionality buttons
         Label func = new Label("Create your Map:");
         Button create = new Button("Create");
-        Button crOpen = new Button("Create & Open");
+        create.setOnAction(e -> {
+            Settings s = saveMap();
+
+            // Write to file (Add number of guards etc.)
+        });
+
+        Button crOpen = new Button("Create & Open"); // Same as above but opens the map also.
 
         vbox.getChildren().addAll(func, create, crOpen);
+
+        Button undo = new Button("Undo");
+        undo.setOnAction(e -> mb.undo());
+
+        vbox.getChildren().addAll(undo);
+    }
+
+    public int parseRect(TextField tf)
+    {
+        try
+        {
+         return Integer.parseInt(tf.getText());
+        }
+        catch(NumberFormatException e)
+        {
+            System.out.println("Input a number");
+        }
+        catch(Exception e)
+        {
+            e.printStackTrace();
+        }
+        return -1;
+    }
+
+    public Settings saveMap()
+    {
```

Comment:
I think the save function might be where you're having the undo issues.  Consider using a stack of UI objects instead?  So every object is drawn and then added to the stack.  If you then undo, all you have to do is pop the last element from the stack and draw a white rectangle there?

Reply:
I think it would be better to not keep the whole settings in this object, rather just the Wall/shade/glass etc objects.  That way we can have another panel that does the Guards/Intruder numbers and the like.  They can all be collected at the point we need the new settings, rather than constantly building new settings objects

Reply:
The issue I see with undoing is if there are two rectangles over each other and you undo one by painting over it white, the one "below" also gets painted over white. Obviously I don't think it makes sense for us to over lap rectangular areas, but maybe an outside user could get confused.

I agree with having a settings panel for the other parts of the setting object and could work on that if you would like. Think it is ready to merge.